### PR TITLE
Add type UserError, use  as indicator.

### PR DESCRIPTION
**Pull request by Eskibear**
~~~"isUserError",  // Indicator of user error. Possible values: "true", "false"~~~

~~~isUserError will be set to true if the error object in instance of UserError~~~


Now using errorType: ErrorType to distinguish "User error" with "System error".

```typescript
enum ErrorType {
//...
//...
}
```

### CODE REVIEW

File: src/next/main.ts [hunk 1]
```diff
@@ -121,10 +122,10 @@ function report(eventName: string, properties?: Properties, measurements?: Measu
...
 
 const ALLOWED_KEYS_FOR_ERROR: string[] = [
-    "errorCode",// Preserve 0 for no error, 1 for general error.
-    "name",     // For name of an Error.
-    "message",  // For message of an Error.
-    "stack",    // For callstack of an Error.
+    "errorCode",    // Preserve "0" for no error, "1" for general error.
+    "message",      // For message of an Error.
+    "stack",        // For callstack of an Error.
+    "isUserError",  // Indicator of user error. Possible values: "true", "false".
```
Comment (akaroml) [inline src/next/main.ts]: You mentioned that boolean values are tricky for dimension entries. If that's still true, then something like `errorType` with the value `userError` or `systemError` makes more sense.

File: src/next/UserError.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export class UserError extends Error {
+    constructor(e: Error) {
+        super(e.message);
+        this.name = e.name;
```
Comment (akaroml) [inline src/next/UserError.ts]: What do these two lines do?

Comment (Eskibear) [reply src/next/UserError.ts]: I find the following definition, but the constructor accepts `message?:string` as the only parameter. I add these lines to copy the values.
```typescript
interface Error {
    message: string;
    name?: string;
    stack?: string;
}
```

File: src/next/main.ts (hunk 1 above)
Comment (Eskibear) [reply src/next/main.ts]: If the opposite side of `userError` is exactly `systemError`, then what you comment here makes sense. Do you think there would be another kind of `nonSystemNonUserError` of which we should take care?

File: src/next/UserError.ts (hunk 2 above)
Comment (Eskibear) [reply src/next/UserError.ts]: I agree these lines look strange. I think following image can explain the reason:
image

What if users want to preserve some extra information in the original error? Is it right to simply discard them when casting it into UserError?

**Comment by Eskibear** [comment]
Merged to "next" branch. Please do code review together in #9

**Pull request merged**